### DISCUSSION TITLE
Installation issue

### DISCUSSION THREAD

**Discussion by Giorgio-C-30** [NatLabRockies/floris#955] [General]
Hi all, 

I have a very easy installation problem. However I am an amateur and I can not solve it.
I have installed floris following the guide with the command "pip install floris". However, when I am trying to run the first script 
"from floris import FlorisModel
fmodel = FlorisModel("gch.yaml") "

I get this error: " No such file or directory: 'gch.yaml' ".
Verifying on the lib folder, I see that only the core folder of floris has been installed, instead of the whole github main folder. Hence the examples folder is missing. I have tried to copy and paste it, but it is not working.

Any suggestion is very welcomed. 
Thank you in advance. 
Giorgio

**Comment by Bartdoekemeijer** [comment]
Hey Giorgio,

If you installed FLORIS through a `pip install floris`, then indeed you may not have the example scripts and files on your system. You can find them here: https://github.com/NREL/floris/tree/main/examples.

You get the error `No such file or directory` because, indeed, you do not have the file `gch.yaml` on your machine (or at least, not in the folder that you are trying to run the Python script out of). You can download `gch.yaml` here: https://github.com/NREL/floris/tree/main/examples/inputs.

Download the `gch.yaml` file and place it in the folder next to your Python script. It should work now.

**Comment by Giorgio-C-30** [comment]
Thank you so much for the quick answer. 
I have copied it in my folder next to my Python script, but i get this error now:

"The filename, directory name, or volume label syntax is incorrect:  attrs generated init floris.core.farm.Farm ".
Any suggestions?

Many thanks,
Giorgio

**Comment by Bartdoekemeijer** [comment]
Hi Giorgio,

See #951 for a more elaborate discussion and a solution to the `<attrs generated init floris.core.farm.Farm>` error.

Cheers,
Bart

**Comment by javadi97** [comment]
hello friends
I have data for a wind farm with six 2 MW turbines, but there is no 2 MW turbine in the Floris library.
Is there a way I can redefine this turbine for Floris?
I want to compare the data obtained from Floris in the field of wake steering with the real wind farm data

**Comment by misi9170** [comment]
Hi @javadi97 , do you have power and thrust curves for your 2MW turbines, and also information on the hub height, rotor diameter, etc? If so, you can generate dictionary representations for them using [`build_cosine_loss_turbine_dict()`](https://github.com/NREL/floris/blob/main/floris/turbine_library/turbine_utilities.py#L10), and then pass the created dictionaries into the [`turbine_type`](https://github.com/NREL/floris/blob/main/floris/floris_model.py#L130) argument of `FlorisModel.set()`

**Reply by misi9170** [reply]
examples/examples_turbines/003_specify_turbine_power_curve.py demonstrates this functionality.

**Comment by misi9170** [comment]
Hi @noorhj,

Thanks for the question--the answer is a bit nuanced. As you know, FLORIS is a steady state model, so does not inherently have a temporal resolution. However, it is also useful to think of FLORIS as modeling the time-averaged flow---but the question remains, what is the time window that the averaging is working over?

Unfortunately, there is not a straightforward answer. We often compare FLORIS to SCADA records from wind farms with ten-minute averaging. This may be more appropriate for small farms, where ten minutes is in the ballpark of the time-scales of the propagation of wind through the farm (for example, if the wind speed is 8 m/s and the wind farm is 2km wide, then the wind may be expected to "traverse" the farm in around 4 minutes). For much larger farms, longer averaging times may be more appropriate.

We also often use the [`UncertainFlorisModel`](https://nrel.github.io/floris/_autosummary/floris.uncertain_floris_model.html#floris.uncertain_floris_model.UncertainFlorisModel), rather than the [`FlorisModel`](https://nrel.github.io/floris/_autosummary/floris.floris_model.html#floris.floris_model.FlorisModel), when we want to consider time averaging. This allows the user to specify standard deviations in the wind direction that may be expected to occur over the averaging window. 3 degrees of standard deviation is usually a good starting point for 10-minute averaging, and I would expect a value slightly higher than that to be best for an hour (maybe 4 or 5 degrees). 

In the end, I think you could use either hourly or ten-minute data. If you want to know more about the `UncertainFlorisModel`, you can take a look at [these examples](https://github.com/NREL/floris/tree/main/examples/examples_uncertain).

I hope this helps!

Misha
